node/src/main/kotlin/net/corda/node/services/vault/NodeVaultService.kt

```
@@ -220,6 +226,175 @@ class NodeVaultService(private val services: ServiceHub, dataSourceProperties: P
         }
     }
 
+    override fun softLockReserve(id: UUID, stateRefs: Set<StateRef>) {
+        if (stateRefs.isNotEmpty()) {
+            val stateRefsAsStr = stateRefsToCompositeKeyStr(stateRefs.toList())
+            // TODO: awaiting support of UPDATE WHERE <Composite key> IN in Requery DSL
+            val updateStatement = """
+                UPDATE VAULT_STATES SET lock_id = '$id', lock_timestamp = '${services.clock.instant()}'
+                WHERE ((transaction_id, output_index) IN ($stateRefsAsStr));
+            """
+            val statement = configuration.jdbcSession().createStatement()
+            log.debug(updateStatement)
+            try {
+                val rs = statement.executeUpdate(updateStatement)
+                if (rs > 0) {
+                    log.trace("Reserving soft lock states for $id: $stateRefs")
+                }
+                statement.close()
+            }
+            catch (e: SQLException) {
+                log.error("""soft lock update error attempting to reserve states: $stateRefs for $id
```

Decided that we will fail with a SoftLockException (extending FlowException) to force end user to handle condition (rather than fail silently and/or return a result code).